Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -242,12 +242,15 @@ class SourceGenerator {
         let groupReference: PBXGroup
 
         if let cachedGroup = groupsByPath[path] {
+            var cachedGroupChildren = cachedGroup.children
             for child in children {
                 // only add the children that aren't already in the cachedGroup
-                if !cachedGroup.children.contains(child) {
-                    cachedGroup.children.append(child)
+                // Check equality by path because XcodeProj.PBXObject.== is very slow.
+                if !cachedGroupChildren.contains(where: { $0.path == child.path }) {
```

Comment:
This _should_ be safe to do. Perhaps we can check for `sourceTree` as well just in case the group has a different root.